app/controllers/incomeSources/add/ForeignPropertyReportingMethodController.scala

```diff
@@ -0,0 +1,257 @@
+        id = id,
+        isAgent = false,
+        itvcErrorHandler = itvcErrorHandler,
+        redirectCall = redirectCall(id),
+        postAction = postAction(id)
+      )
+  }
+
+  def submitAgent(id: String): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            handleSubmitRequest(
+              id = id,
+              isAgent = true,
+              itvcErrorHandler = itvcErrorHandlerAgent,
+              redirectCall = redirectCallAgent(id),
+              postAction = postActionAgent(id)
+            )
+        }
+  }
+
+  def show(id: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        id = id,
+        isAgent = false,
+        itvcErrorHandler = itvcErrorHandler,
+        redirectCall = redirectCall(id),
+        postAction = postAction(id),
+      )
+  }
+
+  def showAgent(id: String): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              id = id,
+              isAgent = true,
+              itvcErrorHandler = itvcErrorHandlerAgent,
+              redirectCall = redirectCallAgent(id),
+              postAction = postActionAgent(id),
+            )
+        }
+  }
+
+  private def handleRequest(id: String,
+                            isAgent: Boolean,
+                            itvcErrorHandler: ShowInternalServerError,
+                            postAction: Call,
+                            redirectCall: Call)
+                           (implicit user: MtdItUser[_]): Future[Result] = {
+
+    if (isEnabled(IncomeSources)) {
+      itsaStatusService.hasMandatedOrVoluntaryStatusCurrentYear.flatMap {
+        case true =>
+
+          val maybeLatencyDetails = user.incomeSources.properties
+            .find(_.incomeSourceId.contains(id))
+            .flatMap(_.latencyDetails)
+
+          maybeLatencyDetails match {
+            case Some(details) =>
+              getForeignPropertyReportingMethodDetails(details).map {
+                case Some(viewModel) =>
+                  Ok(foreignPropertyReportingMethodView(
+                    addForeignPropertyReportingMethodForm = AddForeignPropertyReportingMethodForm.form,
+                    foreignPropertyReportingViewModel = viewModel,
+                    postAction = postAction,
+                    isAgent = isAgent
+                  ))
+                case _ => Redirect(redirectCall)
+              }
+            case _ =>
+              Logger("application").error(s"[ForeignPropertyReportingMethodController][handleRequest]: Failed to retrieve latency details")
```

Comment:
pls add exact error details in the log, there must be some exception associated with this error